api/src/test/java/org/pmiops/workbench/db/dao/InstitutionDaoTest.java

```diff
@@ -3,65 +3,192 @@
 
 @RunWith(SpringRunner.class)
 @DataJpaTest
-@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+@DirtiesContext(classMode = ClassMode.BEFORE_EACH_TEST_METHOD)
 public class InstitutionDaoTest {
+
   @Autowired InstitutionDao institutionDao;
 
-  private DbInstitution testInst = new DbInstitution("Broad", "The Broad Institute");
+  private DbInstitution institutionNoEmailAddressesOrDomains;
+  private DbInstitution institutionWithEmailAddressesAndDomains;
+  private Set<DbInstitutionEmailAddress> emailAddresses;
+  private Set<DbInstitutionEmailDomain> emailDomains;
 
   @Before
   public void setUp() {
-    testInst = institutionDao.save(testInst);
+    institutionNoEmailAddressesOrDomains =
+        institutionDao.save(
+            new DbInstitution().setShortName("Broad").setDisplayName("The Broad Institute"));
+
+    emailAddresses =
+        Sets.newHashSet(new DbInstitutionEmailAddress().setEmailAddress("emailAddress"));
+    emailDomains = Sets.newHashSet(new DbInstitutionEmailDomain().setEmailDomain("emailDomain"));
+
+    institutionWithEmailAddressesAndDomains =
+        institutionDao.save(
+            new DbInstitution()
+                .setShortName("Broad1")
+                .setDisplayName("The Broad Institute")
+                .setEmailAddresses(emailAddresses)
+                .setEmailDomains(emailDomains));
   }
 
   @Test
   public void test_save() {
-    assertThat(institutionDao.findAll()).hasSize(1);
-    institutionDao.save(new DbInstitution("VUMC", "Vanderbilt"));
-    assertThat(institutionDao.findAll()).hasSize(2);
+    final DbInstitution toSaveWith =
+        new DbInstitution()
+            .setShortName("Vanderbilt")
+            .setDisplayName("Vanderbilt University")
+            .setEmailDomains(emailDomains)
+            .setEmailAddresses(emailAddresses);
+    final DbInstitution savedWith = institutionDao.save(toSaveWith);
+    assertThat(savedWith).isEqualTo(toSaveWith);
+    assertThat(savedWith.getEmailDomains()).isEqualTo(emailDomains);
+    assertThat(savedWith.getEmailAddresses()).isEqualTo(emailAddresses);
+
+    final DbInstitution toSaveWithout =
+        new DbInstitution()
+            .setShortName("VUMC")
+            .setDisplayName("Vanderbilt University Medical Center");
+    final DbInstitution savedWithout = institutionDao.save(toSaveWithout);
+    assertThat(savedWithout).isEqualTo(toSaveWithout);
+    assertThat(savedWithout.getEmailDomains()).isEmpty();
+    assertThat(savedWithout.getEmailAddresses()).isEmpty();
   }
 
   @Test
   public void test_delete() {
-    assertThat(institutionDao.findAll()).hasSize(1);
-    institutionDao.delete(institutionDao.findOneByShortName("Broad").get());
-    assertThat(institutionDao.findAll()).hasSize(0);
+    institutionDao.delete(institutionNoEmailAddressesOrDomains.getInstitutionId());
+    DbInstitution dbInstitution =
+        institutionDao.findOne(institutionNoEmailAddressesOrDomains.getInstitutionId());
```

Comment:
nit: if findOne wold return Optional, that'd be great.

Reply:
findOne() is built-in to CrudRepository.
the main accessor for client code findOneByShortName() does return an optional.